### PR TITLE
refactor: add scss, add BAM, change naming

### PR DESCRIPTION
**Pull request by denischernow**

### CODE REVIEW

File: my-app/src/Container/constantsContainer/constantContainer.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import avatar1 from "../../assets/img/chat_avatar_01.jpg";
+import avatar2 from "../../assets/img/chat_avatar_02.jpg";
+import avatar3 from "../../assets/img/chat_avatar_03.jpg";
+import avatar4 from "../../assets/img/chat_avatar_04.jpg";
+import avatar5 from "../../assets/img/chat_avatar_05.jpg";
+
+export const persons = [
```
Comment (DzianisPasiukou) [inline my-app/src/Container/constantsContainer/constantContainer.js:7]: all constants should be in SUCH_STYLE

File: my-app/src/Container/container.js [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+export function Container() {
+	// the code below is required to add the initial message to the localStorage
+	React.useEffect(() => {
+		persons.map((e) => {
+			return localStorage.setItem(e.firstName, firstMessages);
+		});
+	}, persons[0]);
+
+	// the code below is required for the default selection of the interlocutor
+	const [context, setContext] = React.useState(["Patrick", "Black", avatar1]);
```
Comment (DzianisPasiukou) [inline my-app/src/Container/container.js:22]: string to constants

**Review by DzianisPasiukou — CHANGES_REQUESTED** [review]

**Comment by DzianisPasiukou** [comment]
title of the request should be in the same style as commit:
feat: refactor , add scss, add BAM, change naming

File: my-app/src/App.js [hunk 3]
```diff
@@ -1,13 +1,11 @@
...
 
-function App() {
-	return (
-		<div>
-			<Container />
-		</div>
-	);
+export function App() {
+  return (
+    <div>
```
[the remark this answers is not in the document]
Comment (denischernow) [reply my-app/src/App.js:7]: I fixed it

File: my-app/src/App.js [hunk 4]
```diff
@@ -1,13 +1,11 @@
 import React from "react";
```
[the remark this answers is not in the document]
Comment (denischernow) [reply my-app/src/App.js:1]: I will rename everything except the name of the react components

File: my-app/src/App.js [hunk 5]
```diff
@@ -1,13 +1,11 @@
...
-import Container from "./Container/Container";
-import "./App.css";
+import { Container } from "./container/container.js";
+import "./stylesApp/app.scss";
```
[the remark this answers is not in the document]
Comment (denischernow) [reply my-app/src/App.js:3]: corrected

File: my-app/src/Container/chat/input/input.js [hunk 6]
```diff
@@ -0,0 +1,26 @@
+import React from "react";
+import "./stylesInput/input.scss";
```
[the remark this answers is not in the document]
Comment (denischernow) [reply my-app/src/Container/chat/input/input.js:2]: corrected

File: my-app/src/Container/chat/chat.js [hunk 7]
```diff
@@ -0,0 +1,37 @@
+import React from "react";
+import "./stylesChat/chat.scss";
+import { Interlocutor } from "./interlocutor/interlocutor.js";
+import { Message } from "./message/message.js";
+import { Input } from "./input/input.js";
+
+export function Chat({ stateMessageMyself, stateMessageinterlocutor, createCorrespondence }) {
```
[the remark this answers is not in the document]
Comment (denischernow) [reply my-app/src/Container/chat/chat.js:7]: variable names have been replaced with simpler and more concise names, also the names still reflect the meaning of the variables

File: my-app/src/Container/chat/input/stylesInput/input.scss [hunk 8]
```diff
@@ -0,0 +1,27 @@
+.section-input {
+	display: flex;
+	align-items: center;
+	justify-content: space-evenly;
+	width: 100%;
+	height: 20%;
+	background-color: #bacae3;
+	.section-input__input {
+		width: 65%;
```
[the remark this answers is not in the document]
Comment (denischernow) [reply my-app/src/Container/chat/input/stylesInput/input.scss:9]: fixed in the whole document (where found)

File: my-app/src/Container/container.js (hunk 2 above)
Comment (denischernow) [reply my-app/src/Container/container.js:22]: corrected this

File: my-app/src/Container/chat/chat.js [hunk 9]
```diff
@@ -0,0 +1,37 @@
...
+	const messagesEndRef = React.useRef(null);
+	React.useEffect(() => {
+		messagesEndRef.current.scrollIntoView({ behavior: "smooth" });
+	}, [stateMessageinterlocutor]);
+
+	return (
+		<div className="chat">
+			<Interlocutor />
+			<div className="chat__message">
+				{[...stateMessageMyself, ...stateMessageinterlocutor]
```
[the remark this answers is not in the document]
Comment (denischernow) [reply my-app/src/Container/chat/chat.js:19]: I agree with you. done it

File: my-app/src/Container/constantsContainer/constantContainer.js (hunk 1 above)
Comment (denischernow) [reply my-app/src/Container/constantsContainer/constantContainer.js:7]: fixed the error as follows:
avatar4  -->  avatar_4 
firstMessages --> first_messages 
etc.

File: my-app/src/Container/chat/message/message.js [hunk 10]
```diff
@@ -0,0 +1,15 @@
+import React from "react";
+import "./stylesMessage/message.scss";
+
+export  function Message({ children, sender }) {
+  
+  return (
+    <div className="messages">
+      {sender === "myself" ? (
```
[the remark this answers is not in the document]
Comment (denischernow) [reply my-app/src/Container/chat/message/message.js:8]: I agree with you, I divided it into two components

File: my-app/src/App.js [hunk 11]
```diff
@@ -1,13 +1,11 @@
-import React from "react";
```
Comment (DzianisPasiukou) [inline my-app/src/App.js:1]: React can be still with upper case

File: my-app/src/Container/chat/chat.js [hunk 12]
```diff
@@ -0,0 +1,49 @@
+import react, { useCallback, useMemo } from "react";
+import "./styles/chat.scss";
+import { Interlocutor } from "./interlocutor/interlocutor.js";
+import { MessageMyself } from "./messageMyself/messageMylef.js";
+import { MessageInterlocator } from "./messageInterlocator/messageInterlocator.js";
+import { Input } from "./input/input.js";
+
+export function Chat({ myself, iterlocutor, startChat }) {
```
Comment (DzianisPasiukou) [inline my-app/src/Container/chat/chat.js:8]: let's discuss variable names

File: my-app/src/Container/chat/chat.js [hunk 13]
```diff
@@ -0,0 +1,49 @@
...
+	const autoScrolling = react.useRef(null);
+	react.useEffect(() => {
+		autoScrolling.current.scrollIntoView({ behavior: "smooth" });
+	}, [iterlocutor]);
+
+	return (
+		<div className="chat">
+			<Interlocutor />
+			<div className="chat__message">
+				{useMemo(
```
Comment (DzianisPasiukou) [inline my-app/src/Container/chat/chat.js:19]: use it outside of markup

File: my-app/src/Container/chat/interlocutor/styles/interlocutor.scss [hunk 14]
```diff
@@ -0,0 +1,20 @@
+.interlocutor {
+	display: flex;
+	align-items: center;
+	justify-content: space-evenly;
+	width: 100%;
+	height: 20%;
+	background-color: #bacae3;
+
+	.interlocutor__name {
+		background-color: #bacae3;
```
Comment (DzianisPasiukou) [inline my-app/src/Container/chat/interlocutor/styles/interlocutor.scss:10]: looks like too many spaces

File: my-app/src/Container/constants/constants.js [hunk 15]
```diff
@@ -0,0 +1,32 @@
...
+
+export const persons = [
+	{ id: 1, firstName: "Patrick", secondName: "Black", avatar: avatar_1 },
+	{ id: 2, firstName: "Stacy", secondName: "Gray", avatar: avatar_2 },
+	{ id: 3, firstName: "Linda", secondName: "White", avatar: avatar_3 },
+	{ id: 4, firstName: "Tony", secondName: "Freeman", avatar: avatar_4 },
+	{ id: 5, firstName: "Leslie", secondName: "Brown", avatar: avatar_5 },
+];
+
+export const first_messages = `{"0":${JSON.stringify({
```
Comment (DzianisPasiukou) [inline my-app/src/Container/constants/constants.js:15]: FIRST_MESSAGES style (upper case)

File: my-app/src/Container/container.js [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+import "./styles/container.scss";
+import { Persons } from "./persons/persons.js";
+import { persons, first_messages, response_massage } from "./constants/constants.js";
+import { Chat } from "./chat/chat.js";
+import { personsContext } from "./context.js";
+
+export function Container() {
+	// the code below is required to add the initial message to the localStorage
+	react.useEffect(() => {
+		persons.map((e) => {
```
Comment (DzianisPasiukou) [inline my-app/src/Container/container.js:11]: why map?

**Review by DzianisPasiukou — CHANGES_REQUESTED** [review]

File: my-app/src/App.js (hunk 11 above)
Comment (denischernow) [reply my-app/src/App.js:1]: fixed all "react"

File: my-app/src/Container/container.js (hunk 16 above)
Comment (denischernow) [reply my-app/src/Container/container.js:11]: changed to forEach

File: my-app/src/Container/constants/constants.js (hunk 15 above)
Comment (denischernow) [reply my-app/src/Container/constants/constants.js:15]: corrected this inaccuracy

File: my-app/src/Container/chat/interlocutor/styles/interlocutor.scss (hunk 14 above)
Comment (denischernow) [reply my-app/src/Container/chat/interlocutor/styles/interlocutor.scss:10]: checked and corrected indentation throughout the document

File: my-app/src/Container/chat/chat.js (hunk 13 above)
Comment (denischernow) [reply my-app/src/Container/chat/chat.js:19]: fixed, differentiated hook and markup

File: my-app/src/Container/chat/chat.js (hunk 12 above)
Comment (denischernow) [reply my-app/src/Container/chat/chat.js:8]: corrected variable names to be as simple as possible and describing their meaning

**Pull request merged**